src/base/iter.rs

```diff
@@ -119,6 +119,43 @@ macro_rules! iterator {
                 self.size
             }
         }
+
+        impl<'a, N: Scalar, R: Dim, C: Dim, S: 'a + $Storage<N, R, C>> DoubleEndedIterator
+            for $Name<'a, N, R, C, S>
+        {
+            #[inline]
+            fn next_back(&mut self) -> Option<$Ref> {
+                unsafe {
+                    if self.size == 0 {
+                        None
+                    } else {
+                        // Pre-decrement `size` such that it now counts to the
+                        // element we want to return.
+                        self.size -= 1;
+
+                        // Fetch strides
+                        let inner_stride = self.strides.0.value();
+                        let outer_stride = self.strides.1.value();
+                        debug_assert_eq!(outer_stride % inner_stride, 0);
+                        let num_rows = outer_stride / inner_stride;
+
+                        // Compute rows and cols remaining
+                        let outer_remaining = self.size / num_rows;
+                        let inner_remaining = self.size % num_rows;
+
+                        // Compute pointer to last element
+                        let last = self.ptr.offset(
+                            (outer_remaining * outer_stride + inner_remaining * inner_stride)
```

Comment:
Benchmark on my machine shows this to be 300x slower than forward iteration. Is there a way to simplify / optimize the math here?

Reply:
After the `inner_size` fix it is only 7x slower, which is acceptable to me.